WordPress/src/androidTest/java/org/wordpress/android/e2etests/ComposeTests.java

```diff
@@ -0,0 +1,105 @@
+package org.wordpress.android.e2etests;
+
+
+import android.support.test.espresso.contrib.ViewPagerActions;
+import android.support.test.rule.ActivityTestRule;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.wordpress.android.R;
+import org.wordpress.android.e2etests.utils.TestData;
+import org.wordpress.android.e2etests.utils.ToastMatcher;
+import org.wordpress.android.ui.WPLaunchActivity;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.action.ViewActions.closeSoftKeyboard;
+import static android.support.test.espresso.action.ViewActions.replaceText;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+
+public class ComposeTests {
+    @Rule
+    public ActivityTestRule<WPLaunchActivity> mActivityTestRule = new ActivityTestRule<>(WPLaunchActivity.class);
+
+    private TestData mData = new TestData();
+    private String mUserEmail = mData.getEmail();
+    private String mUserPassword = mData.getPassword();
+
+    @Test
+    public void testBlogPosting() {
+        onView(withText("Log In"))
+                .perform(click());
+
+        onView(withId(R.id.input))
+                .perform(replaceText(mUserEmail), closeSoftKeyboard());
+
+        onView(withText("Next"))
+                .perform(click());
+
+        try {
+            Thread.sleep(5000);
```

Comment:
It seems that using `Thread.sleep` is an anti-pattern according to https://developer.android.com/training/testing/espresso/idling-resource.html. Can we use an idling-resource based solution for waiting for the Password screen to come up? Such a solution would help minimize the test flakiness.

Reply:
Ideally we could implement for network calls the https://github.com/JakeWharton/okhttp-idling-resource but needs a bit of overhaul work, or go with https://developer.android.com/reference/android/support/test/espresso/idling/CountingIdlingResource.html perhaps? What do you think @hypest ?

Reply:
Indeed, according to [the recommended avenue](https://developer.android.com/training/testing/espresso/idling-resource.html#integrate-into-app) using an idling resource will touch production code too, but that feels like a good investment to make, especially since we'd like to build and expand the UI tests on a reliable base.

In that respect, the `CountingIdlingResource` looks like a good start, yeah. 👍